flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/SchemaOperator.java

```diff
@@ -111,8 +115,13 @@ private SchemaChangeResponse requestSchemaChange(
         return sendRequestToCoordinator(new SchemaChangeRequest(tableId, schemaChangeEvent));
     }
 
-    private ReleaseUpstreamResponse requestReleaseUpstream() {
-        return sendRequestToCoordinator(new ReleaseUpstreamRequest());
+    private void requestReleaseUpstream() throws InterruptedException {
+        CoordinationResponse coordinationResponse =
+                sendRequestToCoordinator(new ReleaseUpstreamRequest());
+        while (coordinationResponse instanceof SchemaChangeProcessingResponse) {
```

Comment:
Continuous loop maybe not acceptable as this may lead endless coordination. Could we introduce a timeout config option with a reasonable default value for coordination ?

Reply:
Done, add a new pipeline config https://github.com/apache/flink-cdc/pull/3128/files#diff-efc0e20a0c40e81813d3be00eadab7bc066598e521b24a85f2c3e5c9b83c2b4bR92 to control this.